app/table/cells/RouterLinkCell.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+
+import { useMemo } from 'react'
+
+import { useApiQuery } from '~/api'
+import { useVpcSelector } from '~/hooks'
+import { Badge } from '~/ui/lib/Badge'
+import { pb } from '~/util/path-builder'
+
+import { EmptyCell, SkeletonCell } from './EmptyCell'
+import { LinkCell } from './LinkCell'
+
+export const RouterLinkCell = ({ value }: { value?: string }) => {
+  const { project, vpc } = useVpcSelector()
+  const { data: subnet, isError } = useApiQuery('vpcRouterView', {
+    path: { router: value || '' },
+    query: { project, vpc },
+  })
+  return useMemo(() => {
+    if (!value) return <EmptyCell />
+    // probably not possible but let’s be safe
+    if (isError) return <Badge color="neutral">Deleted</Badge>
+    if (!subnet) return <SkeletonCell /> // loading
+    return (
+      <LinkCell to={pb.vpcRouter({ project, vpc, router: subnet.name })}>
+        {subnet.name}
+      </LinkCell>
+    )
+  }, [value, project, vpc, isError, subnet])
```

Comment:
No need for `useMemo` here. In the `useCustomRouterItems` case where I recommended it, the problem is that without memoization, the `items` array is new on every render, which means it will always cause a re-render of the component it's passed into. In this case, you're just rendering this cell and nothing is messed up by it not being referentially stable. The result is basically the same as if you didn't have the `useMemo`, so `useMemo` actually _adds_ overhead (however minimal it may be).

> [So when should I useMemo and useCallback?](https://kentcdodds.com/blog/usememo-and-usecallback#so-when-should-i-usememo-and-usecallback)
> 1. Referential equality
> 2. Computationally expensive calculations